hummingbot/client/settings.py

```diff
@@ -29,84 +33,75 @@
-        connectors = [f.name for f in scandir(f'{str(package_dir)}/hummingbot/connector/connector')
-                      if f.is_dir() and f.name not in invalid_names]
-        others.update(connectors)
-    except Exception:
-        pass
-    return others
 
-
-def _get_example_asset(pair=True):
-    pairs = []
-    fetched_connectors = []
-    for connector_type, connectors in ALL_CONNECTORS.items():
-        for connector in connectors:
-            module_path = f"hummingbot.connector.{connector_type}.{connector}.{connector}_utils"
-            try:
-                if pair:
-                    pairs.append(getattr(importlib.import_module(module_path), "EXAMPLE_PAIR"))
-                else:
-                    pairs.append(getattr(importlib.import_module(module_path), "EXAMPLE_PAIR").split("-")[0])
-            except Exception:
+def _create_connector_settings() -> Dict[str, ConnectorSetting]:
+    connector_exceptions = ["paper_trade"]
+    connector_settings = {}
+    package_dir = Path(__file__).resolve().parent.parent.parent
+    type_dirs = [f for f in scandir(f'{str(package_dir)}/hummingbot/connector') if f.is_dir()]
+    for type_dir in type_dirs:
+        connector_dirs = [f for f in scandir(type_dir.path) if f.is_dir()]
+        for connector_dir in connector_dirs:
+            if connector_dir.name.startswith("_") or \
+                    connector_dir.name in connector_exceptions or \
+                    not any(f.name == f"{connector_dir.name}_utils.py" for f in scandir(connector_dir.path)):
                 continue
-            fetched_connectors.append(connector)
-    return dict(zip(fetched_connectors, pairs))
+            if connector_dir.name in connector_settings:
+                raise Exception(f"Multiple connectors with the same {connector_dir.name} name.")
+            path = f"hummingbot.connector.{type_dir.name}.{connector_dir.name}.{connector_dir.name}_utils"
+            util_module = importlib.import_module(path)
+            if util_module is None:
+                raise Exception(f"{path} does not exist.")
+            fee_type = TradeFeeType.Percent
+            fee_type_setting = getattr(util_module, "FEE_TYPE", None)
+            if fee_type_setting is not None:
+                fee_type = TradeFeeType[fee_type_setting]
+            connector_settings[connector_dir.name] = ConnectorSetting(
+                name=connector_dir.name,
+                type=ConnectorType[type_dir.name.capitalize()],
+                centralised=getattr(util_module, "CENTRALIZED", True),
+                example_pair=getattr(util_module, "EXAMPLE_PAIR", ""),
+                use_ethereum_wallet=getattr(util_module, "USE_ETHEREUM_WALLET", False),
+                fee_type=fee_type,
+                fee_token=getattr(util_module, "FEE_TOKEN", ""),
+                default_fees=getattr(util_module, "DEFAULT_FEES", []),
+                config_keys=getattr(util_module, "KEYS", [])
```

Comment:
Also, in addition to the initial comment `config_keys=getattr(util_module, "KEYS", {})`